odoo/addons/base/models/ir_attachment.py

```diff
@@ -436,8 +436,8 @@ def _search(self, args, offset=0, limit=None, order=None, count=False, access_ri
                 continue
             # model_attachments = {res_model: {res_id: set(ids)}}
             model_attachments[row['res_model']][row['res_id']].add(row['id'])
-            # Should not retrieve binary fields attachments
-            if row['res_field']:
+            # Should not retrieve binary fields attachments if not explicitly required
+            if 'res_field' not in [arg[0] for arg in args] and row['res_field']:
```

Comment:
The patch does not make any sense to me, and your description is even more confusing.
The code above (line 409) explicitly adds a condition on `res_field` in `args` if there is none on the fields `id` and `res_field`.
So, in your case, the condition you add in the `if` statement is *always true*, and therefore it does not change anything.

Reply:
I've updated the PR:
- tried to provide a better description
- My bad, I didn't see the code in line 409, anyway my code evaluated to `false`, and this let the results return correctly

I added the variable `res_field_check` to know if the user want to filter on `res_field` , so to avoid omitting those records